src/ReflectionFileNamespace.php

```diff
@@ -357,6 +362,20 @@ private function findConstants()
             }
         }
 
+        if ($withDefinedConstants) {
+            foreach ($this->namespaceNode->stmts as $namespaceLevelNode) {
```

Comment:
I can guess that we need to perform recursive scan here, not just a quick top-list nodes search. Example:

```php
if (!defined('SOME_CONST')) {
    define('SOME_CONST', 42);
}
```

So, maybe we initialize a node visitor here and use it for traversing, instead of direct scan?